template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmHostGroupRoleConfigProviderProcessor.java

```diff
@@ -124,54 +122,48 @@ private void updateConfigsInTemplate(CmTemplateProcessor templateProcessor, Map<
         for (Map.Entry<String, Map<String, List<ApiClusterTemplateConfig>>> entry : newConfigsByRCG.entrySet()) {
             String configGroupName = entry.getKey();
             ApiClusterTemplateRoleConfigGroup configGroup = roleConfigGroupByName.get(configGroupName);
+            ApiClusterTemplateService templateService = serviceByRCG.get(configGroupName);
             Map<String, List<ApiClusterTemplateConfig>> configsByHostGroup = entry.getValue();
             int groupCount = configsByHostGroup.size();
-            switch (groupCount) {
-                case 1:
-                    templateProcessor.mergeRoleConfigs(configGroup, configsByHostGroup.values().iterator().next());
-                    break;
-                case 0:
-                    // noop
-                    break;
-                default:
-                    LOGGER.debug("Cloning config group {} into {} host groups: {}", configGroupName, groupCount, configsByHostGroup.keySet());
-
-                    // "clone" config group for each host group
-                    Map<String, ApiClusterTemplateRoleConfigGroup> clonesByHostGroup = configsByHostGroup.keySet().stream()
-                            .map(hostGroupName -> Pair.of(hostGroupName, copyForHostGroup(configGroup, hostGroupName)))
-                            .collect(toMap(Pair::getKey, Pair::getValue));
-
-                    // add host group-specific configs to clones
-                    configsByHostGroup.forEach((hostGroupName, newConfigs) ->
-                            templateProcessor.mergeRoleConfigs(clonesByHostGroup.get(hostGroupName), newConfigs));
-
-                    // remove original from service
-                    ApiClusterTemplateService service = serviceByRCG.get(configGroupName);
-                    service.getRoleConfigGroups().removeIf(group -> Objects.equals(group.getRefName(), configGroupName));
-
-                    // remove original from instantiator
-                    if (instantiatorRoleConfigGroups != null) {
-                        instantiatorRoleConfigGroups.removeIf(groupInfo -> Objects.equals(groupInfo.getRcgRefName(), configGroupName));
-                    }
+            if (groupCount == 1 || isSharedComponent(templateService.getServiceType(), configGroup.getRoleType())) {
```

Comment:
This is a complex method, I think we shall increase the coverage

Reply:
> This is a complex method, I think we shall increase the test coverage

Test coverage is 91% in this class. I read other tests and the expression is well-tested.